File: src/main/java/co/gatedaccess/web/repo/CommunityRepo.java
package co.gatedaccess.web.repo;

import co.gatedaccess.web.model.Community;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface CommunityRepo extends MongoRepository<Community, String> {

    Community findCommunityById(String id);

    Community findCommunityBySuperAdminId(String id);
}

File: src/main/kotlin/ng/cove/web/data/repo/CommunityRepo.kt
package ng.cove.web.data.repo

import ng.cove.web.data.model.Community
import org.springframework.data.mongodb.repository.MongoRepository

interface CommunityRepo : MongoRepository<Community, String> {
    fun findByAdminsContains(adminId: String): Community?
    fun findCommunityByIdAndAdminsContains(id: String, adminId: String): Community?
    fun existsByAdminsContains(adminId: String): Boolean
}
